forc-plugins/forc-crypto/src/keygen/new_key.rs

```diff
@@ -0,0 +1,60 @@
+//! This file will be hosted here until
+//! https://github.com/FuelLabs/sway/issues/5170 is fixed
+use super::KeyType;
+use anyhow::Result;
+use fuel_core_types::{
+    fuel_crypto::{
+        rand::{prelude::StdRng, SeedableRng},
+        SecretKey,
+    },
+    fuel_tx::Input,
+};
+use libp2p_identity::{secp256k1, Keypair, PeerId};
+use serde_json::json;
+use std::ops::Deref;
+
+/// Generate a random new secret & public key in the format expected by fuel-core
+#[derive(Debug, clap::Args)]
+#[clap(author, version, about = "Creates a new key for use with fuel-core")]
+pub struct Arg {
+    /// Key type to generate. It can either be `block-production` or `p2p`.
+    #[clap(
+        long = "key-type",
+        short = 'k',
+        value_enum,
+        default_value = KeyType::BlockProduction.into(),
+    )]
+    key_type: KeyType,
+}
+
+pub fn handler(arg: Arg) -> Result<serde_json::Value> {
+    let mut rng = StdRng::from_entropy();
+    let secret = SecretKey::random(&mut rng);
+    let public_key = secret.public_key();
+    let secret_str = secret.to_string();
+
+    let output = match arg.key_type {
+        KeyType::BlockProduction => {
+            let address = Input::owner(&public_key);
+            json!({
+                "secret": secret_str,
+                "address": address,
+                "type": <KeyType as std::convert::Into<&'static str>>::into(KeyType::BlockProduction),
```

Comment:
nit: Does `arg.key_type.into()` not work?

Reply:
It does not, I can however do it in a separated statement. 

```
let type: &str = arg.key_type.into();
```

Should I do that?

Reply:
I would personally prefer `let type: &str = arg.key_type.into();` happy with it either way though!